src/p4est_wrap.c

```diff
@@ -319,6 +320,26 @@ p4est_wrap_new_copy (p4est_wrap_t * source, size_t data_size,
     p4est_reset_data (pp->p4est, data_size, NULL, NULL);
   }
 
+  /* copy newly_adapted arrays if set */
+  P4EST_ASSERT (source->params.store_adapted ||
```

Comment:
Is it not possible that the arrays are non-NULL and afterwards the parameter store_adapted is changed from 1 to 0? On the other hand, parameters are no longer changed during the life-time of the wrap object officially, but unofficial uses may exist.

Reply:
We may adopt the following principle: When working with the stored arrays, we query them for being NULL or not and generally ignore the parameter. Only when deciding to re-allocate them after adaptation do we query the parameter. Would that make sense?

Then the only assertion that we may need is (newly_refined == NULL) == (newly_coarsened == NULL).

Reply:
This seems like the most consistent convention, I have updated the code accordingly.